Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -339,14 +354,15 @@ void WebFullScreenManager::enterFullScreenForElement(Element& element, HTMLMedia
         m_inWindowFullScreenMode = true;
     } else {
         ASSERT(m_elementFrameIdentifier);
-        m_page->sendWithAsyncReply(Messages::WebFullScreenManagerProxy::EnterFullScreen(*m_elementFrameIdentifier, m_element->document().quirks().blocksReturnToFullscreenFromPictureInPictureQuirk(), WTFMove(mediaDetails)), [
+        m_page->sendWithAsyncReply(Messages::WebFullScreenManagerProxy::EnterFullScreen(*m_elementFrameIdentifier, m_element->document().quirks().blocksReturnToFullscreenFromPictureInPictureQuirk(), WTFMove(mediaDetails), rootViewRectOfContents(element)), [
             this,
             protectedThis = Ref { *this },
             element = Ref { element },
             willEnterFullScreenCallback = WTFMove(willEnterFullScreenCallback),
             didEnterFullScreenCallback = WTFMove(didEnterFullScreenCallback)
-        ] (bool success) mutable {
-            if (success) {
+        ] (std::optional<WebCore::IntRect> transformedInitialFrame) mutable {
+            if (transformedInitialFrame) {
+                m_initialFrame = *transformedInitialFrame;
```

Comment:
Ideally this would replace like 321 where we set m_initialFrame to screenRectOfContents(element).  Also, we should probably rename the variable to make clear what coordinates the frame is in.